apiserver/Makefile

```diff
@@ -1,8 +1,31 @@
 .PHONY: install-hail-locally build push run run-hail rm deploy build-hail-shadow-jar
 
-PROJECT = $(shell gcloud config get-value project)
+PROJECT := $(shell gcloud config get-value project)
 GSA_KEY_FILE = /gsa-key/privateKeyData
 
+ifeq ($(IN_HAIL_CI),1)
+define DOCKER_IMAGE
+build-$(1): $(2)
+	-docker pull gcr.io/$(PROJECT)/$(1):latest
+	-docker pull $$(shell grep FROM Dockerfile.$(1) | head -n 1 | sed 's/^HEAD //')
+	docker build -t $(1) -f Dockerfile.$(1) . --cache-from $(1),gcr.io/$(PROJECT)/$(1):latest,$$(shell grep FROM Dockerfile.$(1) | head -n 1 | sed 's/^HEAD //')
```

Comment:
I'm not quite sure what the grep regex does.

If I test it in shell, it return "From hail-base" for Docker.apiserver. I think the intention is to include the image pulled. If so, this would work:

```sh
head -n 1 Dockerfile.($1) | sed 's/FROM //'
```

Reply:
`Dockerfile`s need not start with the `FROM` line, so neither of these work in general.

But yeah, the sed should say `FROM`, I've fixed that.

Reply:
Hmm, I guess it's a little nuanced. The docs say the Dockerfile must start with a `FROM` but in practice you can have commented lines and blank lines before a `FROM` and `docker build` is OK.

Reply:
Thanks for the background! We could probably standardize our implementations to avoid commented out blank lines.